examples/nxp/rt1020-evk-make-baremetal-builtin/main.c

```diff
@@ -31,41 +31,57 @@ static void timer_fn(void *arg) {
            ifp->ndrop, ifp->nerr));
 }
 
-#ifndef RUNINFLASH
-static void fn(struct mg_connection *c, int ev, void *ev_data) {
-  struct mg_tcpip_if *ifp = (struct mg_tcpip_if *) c->fn_data;
-  if (ev == MG_EV_HTTP_MSG) {
-    struct mg_http_message *hm = (struct mg_http_message *) ev_data;
-    if (mg_http_match_uri(hm, "/api/hello")) {  // Request to /api/hello
-      mg_http_reply(c, 200, "", "{%m:%u,%m:%u,%m:%u,%m:%u,%m:%u}\n",
-                    MG_ESC("eth"), ifp->state, MG_ESC("frames_received"),
-                    ifp->nrecv, MG_ESC("frames_sent"), ifp->nsent,
-                    MG_ESC("frames_dropped"), ifp->ndrop,
-                    MG_ESC("interface_errors"), ifp->nerr);
-    } else if (mg_http_match_uri(hm, "/")) {  // Index page
-      mg_http_reply(
-          c, 200, "", "%s",
-          "<html><head><link rel='icon' href='data:;base64,='></head><body>"
-          "<h1>Welcome to Mongoose</h1>"
-          "See <a href=/api/hello>/api/hello</a> for REST example"
-          "</body></html>");
-    } else {  // All other URIs
-      mg_http_reply(c, 404, "", "Not Found\n");
-    }
-  }
+#ifdef MQTT_DASHBOARD
```

Comment:
Not in this PR:
We have those no-op stubs defined for may projects that don't have / don't use real hardware.
I suggest to move these no-op stubs to the mqtt-dashboard/net.c itself - that would be a default implementation, no-op. Wrap it into conditionals to provide an ability  to redefine.

mqtt-dashboard/net.c:

```c
#ifndef ENABLE_CUSTOM_HAL
bool hal_gpio_write(....) {
  MG_DEBUG(("pin: .., val:...", ...));
}
...
#endif
```